### DISCUSSION TITLE
Light theme not working on windows?

### DISCUSSION THREAD

**Discussion by gabrielcarloto** [johnlindquist/kit#1008] [Errors]
Don't know if I missed something, but setting `appearance` to `auto` or `light` in the settings should work, right? It actually just changes the text colors in code blocks:

![image](https://user-images.githubusercontent.com/68694907/213441396-0058d436-73c3-4ddf-9a06-060992e46f58.png)
![image](https://user-images.githubusercontent.com/68694907/213442253-7cd7e43f-e1d8-4dca-a49e-57018331c835.png)

I know custom themes is a pro feature, but does this include the light theme?

**Comment by johnlindquist** [comment]
For now, yes. I removed "automatic" theming based on the OS appearance. It was causing some complicated issues:

- There were way more people demanding the dark theme when the OS appearance was "light"
- When appearance is "auto", OSX bases the appearance on your desktop wallpaper making auto-theming strange with multiple monitors
- Had to manage twice as much CSS for dark/light modes
- It was very, very, very, very, very, very, very annoying to test. Like really super bad. Think of something awful, then multiply it by a million.

I did a major refactor that stripped out all the dark/light stuff, and now it's just a single theme regardless of appearance.

Now I just have to dive back in and make a few of the themes (including light) free. I'll try to get around to that soon.

**Reply by johnlindquist** [reply]
Actually, if you want to "hack" it back in:

![CleanShot 2023-01-19 at 08 44 51](https://user-images.githubusercontent.com/36073/213487596-08b488fc-1b8d-4d6c-9e25-1d5a9671ae44.png)

Then run this in the console:

```js
let theme = {
  "--opacity": "0.85",
  "--color-text": "44, 44, 44",
  "--color-primary": "47, 134, 211",
  "--color-contrast": "47, 134, 211",
  "--color-secondary": "204, 204, 240",
  "--color-background": "255,255,255",
}

Object.entries(theme).forEach(([key, value]) => {
  document.documentElement.style.setProperty(key, value)
})
```

**Reply by Ambushfall** [reply]
#### An easier workaround until the whole theme matter is settled exists and you can use it inside of scripts by calling the setScriptTheme Method.

theme-color-picker.ts
```ts
// Name: Theme Maker
// Description: Test Themes
// @ts-nocheck

import "@johnlindquist/kit"

interface ThemeObj {
    name: string;
    type: string;
    input: string;
}

const setArgOptions = (key: string, val: string) : ThemeObj => { return { name: key, type: "color", input: val } }

// name: "foreground" | "accent" | "ui" | "background"
// type: "color"
// input: `${defaultValue}` @opacity 0 - 100

let foreground = await arg(setArgOptions("foreground", "#ffffff"))
let accent = await arg(setArgOptions("accent", "#fbbf24"))
let ui = await arg(setArgOptions("ui", "#343434"))
let background = await arg(setArgOptions("background", "#000000"));
let opacity = await arg(setArgOptions("opacity","85"))

let opacitystr = `0.${opacity}`

setScriptTheme({
    foreground: foreground,
    accent: accent,
    background: background,
    opacity: opacitystr,
    ui: ui,
    text: ""
})

let result = await div(md(`# Success!`))
```
